File: backend/app/services/cache_service.py
import json
import redis
from app.core.config import settings


class CacheService:

    def __init__(self):
        self.client = redis.Redis.from_url(settings.REDIS_URL, decode_responses=True)

    def get(self, key: str):
        try:
            value = self.client.get(key)
            if value:
                print("⚡ Cache hit:", key)
                return json.loads(value)
        except Exception as e:
            print("Redis get failed:", e)
        return None

    def set(self, key: str, value: dict, ttl: int = 300):
        try:
            self.client.setex(
                key,
                ttl,
                json.dumps(value)
            )
            print("💾 Cached:", key)
        except Exception as e:
            print("Redis set failed:", e)

File: backend/src/services/cacheService.js
const redis = require("redis");

const client = redis.createClient({
  url: process.env.REDIS_URL || "redis://localhost:6379"
});

client.connect().catch(console.error);

async function getCache(key) {
  try {
    const data = await client.get(key);
    return data ? JSON.parse(data) : null;
  } catch {
    return null;
  }
}

async function setCache(key, value, ttl = 300) {
  try {
    await client.setEx(key, ttl, JSON.stringify(value));
  } catch {}
}

module.exports = { getCache, setCache };